### PR TITLE
Fix building with conda-forge due to the layout change

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: torch-common.sh [hunk 1]
```diff
@@ -43,10 +43,13 @@ export USE_MEM_EFF_ATTENTION=0
 export CMAKE_PREFIX_PATH=$CONDA_PREFIX
 
 # Use cudatoolkit from conda (see pytorch-dev.yaml)
-# If you have it installed system-wide (e.g. in qgpu) point CUDA_PATH to the right folder
-export CUDA_PATH=$CONDA_PREFIX
+# If you have it installed system-wide (e.g. in qgpu) point CUDA_PATH and CMAKE_CUDA_COMPILER
+# to the right folder
+# Note: targets/x86_64-linux is added because of the use of deprecated find_package(CUDA)
+# and layout change in conda
```
Comment (lezcano) [inline torch-common.sh:49]: Mind explaining this a bit more?

**Review by lezcano — APPROVED** [review]
Thank you!

By the way, if you think that you'd be able to put together a PR updating find_package(CUDA) to enable_langauges(CUDA) in PyTorch core, that'd be a very cool contribution. @peterbell10 has done some work on cmake in PyTorch, so he may have more context on whether this can be achieved and how (I don't know much about modern cmake really)

**Comment by isuruf** [comment]
It uses both and checks that the two agree at https://github.com/pytorch/pytorch/blob/main/cmake/public/cuda.cmake#L63-L68. Not sure why both are used.

**Comment by peterbell10** [comment]
PyTorch itself only uses `enable_language(CUDA)` but some of our dependencies still use `find_package(CUDA)` so I left that call in place so we can check that the two give consistent results, otherwise you could potentially run into some weird bugs where two different versions of CUDA are used.

Ideally the dependencies would be updated and we could uniformly use the new form everywhere.

**Pull request merged**